_docs/use_functions.md

```diff
@@ -0,0 +1,126 @@
+When a function requires parameters, they must be provided with concrete values.
+
+In the following version of `my_func`, the parameters `param1` and `param2` are assigned the values `42` and `42`:
+
+```glu
+func main() {
+    my_func(42, 42); // This will print 84
+}
+
+func my_func(param1: Int, param2: Int) {
+    std::print(param1 + param2);
+}
+```
+
+This function will print `84` as `my_func` simply adds `param1` to `param2`.
+
+Since the types of `my_func` parameters are explicitly specified as `Int`, you must provide strictly `Int` typed values.
+When declaring a function with parameters, the type of each parameter must be explicitly stated.
+Note that the `my_func` function is defined after the `main` function in the source code, but it could have been defined before.
+Glu does not enforce the order of function definitions, only that they are defined within an accessible scope.
+
+### Optional parameters
+
+Optional parameters can be specified in Glu by specifying default values to any function parameter:
+
+```glu
+func main() {
+    my_func(2); // This will print 44
+}
+
+func my_func(param1: Int, param2: Int = 42) {
+    std::print(param1 + param2);
+}
+```
+
+With this declaration, you do not need to provide `param2` a value, the default one will be used .
+You should keep in mind that the order in which your optional parameters are written have his importance.
+If a non-optional parameter is asked with an optional one, the optional parameter must always be declared last.
+
+## Functions with Return Values
+
+Functions can return values to the code that calls them.
+When declaring a function that returns a value, you must specify the type of the return value after an arrow `->`.
+If a function specifies a return value, the `return` keyword must be used to return the desired value.
+
+```glu
+func add(a: Int, b: Int) -> Int {
+    return a + b;
+}
+
+func main() {
+    let result = add(3, 5);
+
+    std::print(result); // This will print 8
+}
+```
+
+In this example, the `add` function takes two parameters, `a` and `b`, both of type `Int`, and returns an `Int`.
+The `-> Int` syntax indicates that the function returns an integer. Within the function body, the `return` keyword is used to return the sum of `a` and `b`.
+
+
+Also, note that the `return` keyword doesn't have to be at the end of the function.
+
+```glu
+func main() {
+    let x: Int = 42;
+
+    if x == 42 {
+        return 0;
+    }
+
+    return 1;
+}
+```
+
+### Conclusion
+
+Understanding the structure and syntax of functions is crucial in Glu programming.
+Functions enable modular and reusable code, making programs more efficient and easier to manage.
+
+
+Functions are defined with four key components: the `func` keyword, the function name, parameters, and the return type, with the body enclosed in curly brackets.
+Also, parameters are special variables required by some functions, and default values can make those parameters optional.
```

Comment:
special variables? they’re not even variables, they are a constant local binding